### PR TITLE
Allow sending a share email to multiple users

### LINKED ISSUE #19848: Send sharing link to more than one recipient [OC 8.1.3]
### Steps:
1. Activate `Allow users to send mail notification for shared files`
2. Share file/folder via link
3. Try to send the shared link to several users: `mail1@example.org mail2@example.org`

### Expected behavior
Mails are sent like in Owncloud 8.0.6 you can send the shared link to more than one user with this string:
`mail1@example.org mail2@example.org`

### Actual behavior
Error shows up in the logfiles:
````
{"reqId":"1JSL4UiItKyRQo+1vc03","remoteAddr":"ip","app":"sharing","message":"Can't send mail with public link to mail1@example.org, mail2@example.org: Address in mailbox given [mail1@example.org, mail1@example.org] does not comply with RFC 2822, 3.6.2.","level":3,"time":"2015-10-16T21:10:46+00:00","method":"POST","url":"\/oc8\/owncloud\/index.php\/core\/ajax\/share.php"}

````
It seems like it should support more than one users because the interface shows this error
```
Error while sharing
Couldn't send mail to following users: mail1@example.org mail2@example.org
```

### Server configuration
**Apps**
Enabled:
 - activity: 2.0.2
 - bookmarks: 0.4
 - calendar: 0.7.3
 - contacts: 0.4.0.0
 - files: 1.1.10
 - files_pdfviewer: 0.7
 - files_sharing: 0.6.2
 - files_texteditor: 0.4
 - files_trashbin: 0.6.3
 - files_versions: 1.0.6
 - files_videoviewer: 0.1.3
 - firstrunwizard: 1.1
 - galleryplus: 14.1.0
 - mail: 0.2.0
 - news: 6.0.5
 - provisioning_api: 0.2
 - templateeditor: 0.1
 - updater: 0.6

Both mail addresses exist, you can send shared link to each address individually.

Tried this:
`mail1@example.org mail2@example.org`
`mail1@example.org, mail2@example.org`
`mail1@example.org; mail2@example.org`
`<mail1@example.org>, <mail2@example.org>`

**Owncloud**: 8.1.3 (on Ubuntu 14.04/apache/mysql)

This problem was originally reported by one of the users on the forum: https://forum.owncloud.org/viewtopic.php?f=19&t=30954

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #19848

@MorrisJobke @rullzer @LukasReschke 

@karlitschek please approve backport to 8.2.2 and 8.1.5

### CODE REVIEW

File: tests/lib/share/MailNotificationsTest.php [hunk 1]
```diff
@@ -123,7 +123,21 @@ public function testSendLinkShareMailWithoutReplyTo() {
...
 	}
 
-	public function testSendLinkShareMailWithReplyTo() {
+	public function dataSendLinkShareMailWithReplyTo() {
+		return [
+			['lukas@owncloud.com nickvergessen@owncloud.com'],
+			['lukas@owncloud.com,nickvergessen@owncloud.com'],
+			['lukas@owncloud.com, nickvergessen@owncloud.com'],
+			['lukas@owncloud.com;nickvergessen@owncloud.com'],
+			['lukas@owncloud.com; nickvergessen@owncloud.com'],
```
Comment (LukasReschke) [inline tests/lib/share/MailNotificationsTest.php]: Can we have a test for a single recipient?

Comment (nickvergessen) [inline tests/lib/share/MailNotificationsTest.php]: Right, seems like I forgot to git add before committing.
Done

File: lib/private/share/mailnotifications.php [hunk 2]
```diff
@@ -176,10 +176,12 @@ public function sendLinkShareMail($recipient, $filename, $link, $expiration) {
 		$subject = (string)$this->l->t('%s shared »%s« with you', [$this->senderDisplayName, $filename]);
 		list($htmlBody, $textBody) = $this->createMailBody($filename, $link, $expiration);
 
+		$recipient = str_replace([', ', '; ', ',', ';', ' '], ',', $recipient);
```
Comment (DeepDiver1975) [inline lib/private/share/mailnotifications.php]: Can an email look like "Dr. Foo, Bar" <foo@bar.com>?

If yes - this will break apart

Comment (nickvergessen) [inline lib/private/share/mailnotifications.php]: Before this patch:
> Die E-Mail konnte nicht an folgende Benutzer gesendet werden: "Dr. Foo, Bar" <nickvergessen@owncloud.com>

We would need to split the email and user name into key => value pairs first:
https://github.com/owncloud/core/blob/a881218fc25c1f650e7b381e3acf2007fdc45c38/lib/private/mail/message.php#L117-L128

we might want to do that one day, but that's a separate issue in my eyes

**Comment by MorrisJobke** [comment]
Tested and works :+1:

**Comment by karlitschek** [comment]
nice. please backport :+1:

**Pull request merged**